Allow multiple random cohorts.

cahrens opened the pull request:
## Continuation of [TNL-4355](https://openedx.atlassian.net/browse/TNL-4355)

Allow multiple random cohorts

### Sandbox
- [x] https://verifiedcohort.sandbox.edx.org

Notes: Demo course has feature fully enabled. You can enroll/upgrade to verified/and unenroll, and the cohorts should update accordingly (to see cohort assignments, go to https://verifiedcohort.sandbox.edx.org/courses/course-v1:edX+DemoX+Demo_Course/instructor#view-data_download as staff and click "List enrolled students' profile information".

The Demo course has 2 random cohorts and 2 manual cohorts configured. When in the audit track, a learner is randomly assigned to one of the 2 random cohorts (Audit 1 and Audit 2)-- both are linked to the Audit content group. When enrolling in the verified track, a learner is automatically moved to the manual cohort "Verified Learners". The "Stanford" manual cohort would be managed strictly by course staff.

The very first unit in the course has 1 HTML component visible to all, 1 visible only to Audit content group, and 1 visible only to Verified content group.

To view logs, grant yourself ssh access to the sandbox (http://jenkins.edx.org:8080/view/Ansible/job/ansible-grant-ssh-access/) , ssh to verifiedcohort.sandbox.edx.org, and look at these 2 logs:
sudo less /edx/var/log/lms/edx.log
sudo less /edx/var/log/supervisor/lms_default_1-stderr.log

To access the admin site (to enable automatic cohorting for a course):
1) Log in to the LMS as user "christina@example.com" with password "edx"
2) Go to this URL: https://verifiedcohort.sandbox.edx.org/admin/verified_track_content/verifiedtrackcohortedcourse/

### Testing
- [x] Unit, integration, acceptance tests as appropriate

### Reviewers
If you've been tagged for review, please check your corresponding box once you've given the :+1:.
- [ ] Code review: @dianakhuang 
- [x] Code review: @efischer19 
- [ ] Product review: @scottrish 

### Post-review
- [ ] Squash commits

cahrens commented:
FYI @catong and @chris-mike this sandbox has our final iteration of the automatic cohorting.

efischer19 commented on lms/djangoapps/verified_track_content/models.py:
```diff
@@ -39,30 +39,24 @@ def move_to_verified_cohort(sender, instance, **kwargs):  # pylint: disable=unus
+                # Note that calling this method will create a "Default Group" random cohort if no random
+                # cohort yet exist.
                 random_cohort = get_random_cohort(course_key)
-                if not is_default_cohort(random_cohort):
```
Should we also update the other usage of `is_default_cohort` to allow for multiple random cohorts? According to my grep, it's only used in this file and one other.
- https://github.com/edx/edx-platform/blob/master/openedx/core/djangoapps/course_groups/cohorts.py#L433-L434 Update error string to say "at least one"
- https://github.com/edx/edx-platform/blob/master/openedx/core/djangoapps/course_groups/cohorts.py#L452-L458 Get all random cohorts, check `len >= 1`, and `name in ` instead of `name ==`

At any rate, the import in line 15 of this file can be removed.

efischer19 commented:
One question about how extensively we want to address the `is_default_cohort` method, we talked about it. Whatever you decide is a-ok by me.

:+1:

cahrens commented:
@efischer19 Thanks, I removed the unused import and renamed the method to make its purpose clearer.

@dianakhuang Please review when you are able... but before tomorrow morning. :)

dianakhuang commented:
:+1: from me too.
